jni/src/faiss_index_service.cpp

```diff
@@ -104,10 +206,113 @@ void IndexService::createIndex(
 
     // Write the index to disk
     faissMethods->writeIndex(idMap.get(), indexPath.c_str());
+    delete idMap->index;
 }
 
 BinaryIndexService::BinaryIndexService(std::unique_ptr<FaissMethods> faissMethods) : IndexService(std::move(faissMethods)) {}
 
+jlong BinaryIndexService::initIndex(
+        knn_jni::JNIUtilInterface * jniUtil,
+        JNIEnv * env,
+        faiss::MetricType metric,
+        std::string indexDescription,
+        int dim,
+        int numVectors,
+        int threadCount,
+        std::unordered_map<std::string, jobject> parameters
+    ) {
+    // Create index using Faiss factory method
+    std::unique_ptr<faiss::IndexBinary> indexWriter(faissMethods->indexBinaryFactory(dim, indexDescription.c_str()));
+
+    // Set thread count if it is passed in as a parameter. Setting this variable will only impact the current thread
+    if(threadCount != 0) {
+        omp_set_num_threads(threadCount);
+    }
```

Comment:
This was originally a remnant of copy paste, and I am not sure if any faiss indices use OpenMP to initialize. I could be wrong, though, and have kept it until now in case there is an index that uses OpenMP to initialize.

Reply:
Resolving this for now since it doesn't affect program flow and could be useful